apps/full-site-editing/full-site-editing-plugin/blocks/page-content/index.js

```diff
@@ -0,0 +1,128 @@
+/* eslint-disable wpcalypso/jsx-classname-namespace */
+/**
+ * External dependencies
+ */
+import classNames from 'classnames';
+import { find, get, map } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls } from '@wordpress/block-editor';
+import { registerBlockType } from '@wordpress/blocks';
+import { IconButton, Placeholder, SelectControl, Toolbar } from '@wordpress/components';
+import { compose, withState } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { Fragment } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+const edit = compose(
+	withSelect( select => ( {
+		pages: select( 'core' ).getEntityRecords( 'postType', 'page', {
+			per_page: -1,
+		} ),
```

Comment:
core already makes this mistake, but this line loads in every page on a site. if you load this in the FieldGuide it will download over 40 MB of data before returning `pages` due to the middleware which follows paged results

Reply:
I know, I planned to replace this whole thing with a modified version of `URLInput` (inspired by your plugin), but decided to avoid doing so now, and improve on it in a follow up PR if/when the dev env gets approved.